### PR TITLE
`Development`: Modify the IntelliJ editor configuration to discourage star imports

### PR DESCRIPTION
**Pull request by reschandreas**
#### General

- [x] This is a small issue that I tested locally in my IDE.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

We have multiple code scanning tools that complain about our usage of star imports or other formatting issues, making it hard to spot any real findings that should be taken care of.

### Description
I exported the default editor configuration of Intellij and removed everything I could see that did not make sense for us. The only values that I changed is the usage of star imports to 1000 to prevent teamscale findings and coderabbit comments.
I **do not** intend to tell you how to format your code, I'd very much like this editor config to be a living document and be changeable by all. After all its just formatting and easily changeable.

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

## Summary by CodeRabbit

- **Refactor**
	- Updated the `.editorconfig` to improve code formatting standards, including line length and tab width adjustments.
- **Chores**
	- Added specific `.editorconfig` settings for Liquibase changelog files to prevent reformatting issues.
	- Included a new `.editorconfig` in the test resources to address potential formatting-related test failures.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -10,17 +10,659 @@ root = true
...
+ij_java_block_comment_at_first_column = true
+ij_java_builder_methods =
+ij_java_call_parameters_new_line_after_left_paren = false
+ij_java_call_parameters_right_paren_on_new_line = false
+ij_java_call_parameters_wrap = off
+ij_java_case_statement_on_separate_line = true
+ij_java_catch_on_new_line = false
+ij_java_class_annotation_wrap = split_into_lines
+ij_java_class_brace_style = end_of_line
+ij_java_class_count_to_use_import_on_demand = 1000
```
Comment (Hialus) [inline .editorconfig:87]: We've previously decided to intentionally set this to 4. Is there any specific reason why you change this here (except for the Teamscale warnings)?

Comment (Hialus) [reply .editorconfig:87]: Especially considering the insane amount of changes this little change introduces, I would at the very least not apply this change to every existing file.
As it stands right now this PR is effectively unreviewable.

Comment (reschandreas) [reply .editorconfig:87]: I wanted to bring this to the dev meeting today, I think if we have multiple tools that comment on every change and complain about star imports, maybe we should simply address it and make teamscale and this new coderabbit actually useful as right now teamscale issues are simply dismissed. 

I also don't intend to merge this PR as-is, hence the draft, I would rather remove the changes again and simply merge the `.editorconfig` I snatched from Intellij and increase the limit as I see now advantage in it being 4, but I'm asking that later and we can discuss.

**Review by nityanandaz — APPROVED** [review]
Thanks for handling the issue

File: .editorconfig [hunk 2]
```diff
@@ -15,12 +15,376 @@ insert_final_newline = true
...
+ij_java_doc_indent_on_continuation = true
+ij_java_doc_keep_empty_lines = true
+ij_java_doc_keep_empty_parameter_tag = true
+ij_java_doc_keep_empty_return_tag = true
+ij_java_doc_keep_empty_throws_tag = true
+ij_java_doc_keep_invalid_tags = true
+ij_java_doc_param_description_on_new_line = false
+ij_java_doc_preserve_line_breaks = false
+ij_java_doc_use_throws_not_exception_tag = true
+ij_java_else_on_new_line = false
```
Comment (Hialus) [inline .editorconfig:113]: This should be true, according to our current spotless settings.

File: .editorconfig [hunk 3]
```diff
@@ -15,12 +15,376 @@ insert_final_newline = true
...
+ij_java_block_comment_add_space = false
+ij_java_block_comment_at_first_column = true
+ij_java_call_parameters_new_line_after_left_paren = false
+ij_java_call_parameters_right_paren_on_new_line = false
+ij_java_call_parameters_wrap = off
+ij_java_case_statement_on_separate_line = true
+ij_java_catch_on_new_line = false
+ij_java_class_annotation_wrap = split_into_lines
+ij_java_class_brace_style = end_of_line
+ij_java_class_count_to_use_import_on_demand = 100
```
Comment (Hialus) [inline .editorconfig:88]: This should ideally be included in Spotless if possible. A quick search suggests that this should be possible.

**Review by Hialus — CHANGES_REQUESTED** [review]
The amount of changes in this file is simply way too big to properly review them. I found at least one instance where it is not in line with our existing Spotless settings. For changes like this we would have to make sure that they are 100% in line with our other linter settings that we use. For this we would have to painstakingly go through each setting and compare them with Spotless (and possibly other linters)
There are several other existing differences between IntelliJ default settings and our Linter settings that I am aware of that likely are also included in the changes here. (e.g. double vs. single quotes in JS code is a common thing).

I think our priority should not be to enforce the default IntelliJ settings in our editorconfig, but to enforce our custom linter rules that differ from the default IntelliJ settings.
It is important to remember that the editorconfig is used when you reformat or generate code in IntelliJ. Our linters run as a pre-commit hook and would then complain about or revert differences.

**Comment by reschandreas** [comment]
Hi @Hialus,
Thanks for the input.
1. I wasn't entirely aware that we are using Spotless for formatting; I knew that we use pre-commit hooks but didn't look into the tool behind it. I assumed the `editorconfig` was enforced - it turns out I was wrong.
2. I don't think this version of the editorconfig will collide with our JS Linters, as I only added stuff for Java and Python; the `[*.java]` or `[*.py]` headers inside the file indicate that.
3. IMHO We should not necessarily stick to custom linter rules; I see no benefit in that. We probably use Teamscale for a reason. If we don't want to ignore ca. [1700 bad practice](https://teamscale.io/findings.html#list/GitHub-ls1intum-Artemis/?t=develop%3AHEAD&filter=Comprehensibility%2FBad+Practice&assessment-filters-empty=true&included-paths-empty=true&excluded-paths-empty=true&guideline-rules-empty=true&invert=true) findings, then either we need to revisit the custom rules or make every linting tool (spotless, coderabbit, teamscale, etc.) aware of those, right now actual findings go unnoticed because of the vast amount of false-positives.

What would you say if I revert all of the lines apart from the increased star import limit (as established in the last meeting, we want to eliminate the magic number of 4, my original intention with this PR) and thus hopefully make Teamscale helpful?

What do you think?

File: .editorconfig (hunk 3 above)
Comment (reschandreas) [reply .editorconfig]: I couldn't quickly find anything that suggests that spotless could do that, I think spotless only looks at one file at a time and does not know which classes of the package are actually used.

**Comment by reschandreas** [comment]
@Hialus, I removed everything except the changes to the usage of star imports. I also aligned the import order specified by Spotless. Now, if you optimize the imports using IntelliJ, it will expand them and remove the lines between the groups (didn't find anything for that). Spotless will validate the imports and readd the blank lines in between. I also couldn't find a setting for Spotless to replace star imports, but I guess with this setting, our code should produce fewer false positives.

**Comment by Strohgelaender** [comment]
I'm not sure if I understood the changes / goals right. My current understanding is the following:

The goal of this PR is to change our approach to star imports in order to get rid of teamscale and coderabbit warnings.
You accived this by changing the editorconfig threshold for start imports, but without touching the spotless configuration.

This means that if I change a file with star imports in one of my PRs, nothing will happen for now, and no tool will complain. The star imports will only get removed when using the "optimize imports" feature of IntelliJ. 

At least in my workflow, the IntelliJ formatter does not play a role. If I want to reformat the code, I quickly execute spotless. 

This means that in this form this change will create an inconsistent code style, depending on the current workflow of the developer. I'd like to avoid this.

Please check if you can enforce star import rules via spotless. Then violations will be automatically corrected in the pre-commit-hook and would lead to a failure in the `server-style` check. The editorconfig file in its current form is not meant to define its own rules, but instead to translate existing rules to the IDE which will then reduce the changes spotless needs to perform.

Regarding the PR size: If the configuration change is small (e.g. only one changed rule), I'm ok with bigger PRs (considering LOC) which enures a consistent code base. But I also agree with Timor, we should touch only 1-2 rules in such PRs to keep them reviewable.

**Comment by reschandreas** [comment]
Hi @Strohgelaender,

sorry I didn't reply earlier.

You're right. My goal/idea is to gradually phase out star imports as they keep causing false positives and leading to a complete neglect of real findings.

As I couldn't find a way to use Spotless only for changed files of a commit, I would rather merge these changes that prevent new ones and after some time, apply the config for all files that are left. This would decrease the potential merge conflicts and developer time for those changes. 

I agree that this results in a temporarily inconsistent code styling, but after all, it's just imports

**Review by Strohgelaender — APPROVED** [review]
As discussied I'm ok with this approach. In a few months we should open another PR making the star import usage consistent and also implement the custom rule mentioned above.

**Review by laurenzfb — APPROVED** [review]

**Pull request merged**